graphics/palette.h

```diff
@@ -111,9 +111,98 @@ class PaletteManager : Common::NonCopyable {
 
 namespace Graphics {
 
+struct Palette {
+	byte data[256 * 3];
+	uint size;
+
+	/**
+	 * @brief Construct a new Palette object
+	 *
+	 * @param num   the number of palette entries
+	 */
+	Palette(uint num = 0);
+
+	Palette(const Palette &p);
+
+	bool equals(const Palette &p) const;
+
+	bool contains(const Palette &p) const;
+
+	void clear();
+
+	/**
+	 * Replace the specified range of the palette with new colors.
+	 * The palette entries from 'start' till (start+num-1) will be replaced - so
+	 * a full palette update is accomplished via start=0, num=256.
+	 *
+	 * The palette data is specified in interleaved RGB format. That is, the
+	 * first byte of the memory block 'colors' points at is the red component
+	 * of the first new color; the second byte the green component of the first
+	 * new color; the third byte the blue component, the last byte to the alpha
+	 * (transparency) value. Then the second color starts, and so on. So memory
+	 * looks like this: R1-G1-B1-R2-G2-B2-R3-...
+	 *
+	 * @param colors	the new palette data, in interleaved RGB format
+	 * @param start		the first palette entry to be updated
+	 * @param num		the number of palette entries to be updated
+	 *
+	 * @note It is an error if start+num exceeds 256, behavior is undefined
+	 *       in that case (the backend may ignore it silently or assert).
+	 * @note It is an error if this function gets called when the pixel format
+	 *       in use (the return value of getScreenFormat) has more than one
+	 *       byte per pixel.
```

Comment:
These comments look like they should be updated since the `Palette` structure isn't directly tied to the backend.

Reply:
Ok, I will update comments.